src/database.ts

```diff
@@ -55,24 +60,102 @@ import {
   TransactionRunner,
 } from './transaction-runner';
 
-export interface GetDatabaseOptions {
+import {google as spanner_client} from '../proto/spanner';
+import {Any, Schema} from './common';
+import {ServiceError, CallOptions} from 'grpc';
+import {Readable, Transform} from 'stream';
+import {Stats} from 'fs';
+import {StreamProxy} from 'google-gax/build/src/streamingCalls/streaming';
+
+interface CreateBatchTransactionCallback {
+  (
+    err?: Error | null,
+    transaction?: BatchTransaction | null,
+    apiResponse?:
+      | spanner_client.spanner.v1.ITransaction
+      | spanner_client.spanner.v1.ISession
+  ): void;
+}
+
+interface GetDatabaseOptions {
   autoCreate?: boolean;
 }
-export type DatabaseResponse = [Database, r.Response];
-export interface DatabaseCallback {
+type DatabaseResponse = [Database, r.Response];
+interface DatabaseCallback {
   (err: Error | null, database?: Database, apiResponse?: r.Response): void;
 }
 
-export interface GetSnapshotCallback {
+interface GetSnapshotCallback {
   (err: Error, snapshot?: null): void;
   (err: null, snapshot: Snapshot): void;
 }
 
-export interface GetTransactionCallback {
-  (err: Error, transaction?: null): void;
-  (err: null, transaction: Transaction): void;
+interface GetTransactionCallback {
+  (err?: Error | null, transaction?: Transaction | null): void;
+}
+
+interface SessionPoolCtor {
+  (arg0: Database, arg1: SessionPoolOptions | null): void;
+}
+
+interface UpdateSchemaCallback {
+  (
+    err: Error | null,
+    operation: GaxOperation,
+    resp: database_admin_client.longrunning.IOperation
+  ): void;
+}
+
+interface MakePooledConfig {
+  reqOpts: {
+    [key: string]: Any;
+    session?: string;
+  };
+  client?: string;
+  method?: string;
+  gaxOptions: CallOptions;
+}
+
+interface MakePooledRequestCallback {
+  (err: Error | null, args: IArguments | null): void; //IArguments to represent arguments object used
```

Comment:
This is incorrect, `apply()` works the same way as the spread argument, so we won't be passing the `arguments` object back as a single parameter, but rather each argument will be its own parameter.